firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/common/CrashlyticsController.java

```diff
@@ -578,14 +580,15 @@ private void doOpenSession(String sessionIdentifier, Boolean isOnDemand) {
   }
 
   void doCloseSessions(SettingsProvider settingsProvider) {
-    doCloseSessions(false, settingsProvider);
+    doCloseSessions(false, settingsProvider, false);
   }
 
   /**
-   * Not synchronized/locked. Must be executed from the single thread executor service used by this
-   * class.
+   *
+   * Not synchronized/locked. Must be executed from the executor service runs tasks in serial order
```

Comment:
Can you add a precondition to enforce this?

Reply:
Done!